cmd/deploy/wait.go

```diff
@@ -66,48 +73,22 @@ func (dw *Waiter) wait(ctx context.Context, opts *Options) error {
 	return nil
 }
 
-func (dw *Waiter) waitForResourcesToBeRunning(ctx context.Context, opts *Options) error {
-	ticker := time.NewTicker(5 * time.Second)
-	to := time.NewTicker(opts.Timeout)
+func (dw *Waiter) waitForResourcesToBeRunning(ctx context.Context, opts *Options, namespace string) error {
 	c, _, err := dw.K8sClientProvider.ProvideWithLogger(okteto.GetContext().Cfg, dw.K8sLogger)
 	if err != nil {
 		return err
 	}
 
-	for {
-		select {
-		case <-to.C:
-			return fmt.Errorf("'%s' resources where not healthy after %s", opts.Manifest.Name, opts.Timeout.String())
-		case <-ticker.C:
-			ns := okteto.GetContext().Namespace
-			dList, err := pipeline.ListDeployments(ctx, opts.Manifest.Name, ns, c)
-			if err != nil {
-				return err
-			}
-			areAllRunning := true
-			for i := range dList {
-				d := &dList[i]
-				if !deployments.IsRunning(ctx, ns, d.Name, c) {
-					areAllRunning = false
-				}
-			}
-			if !areAllRunning {
-				continue
-			}
-			sfsList, err := pipeline.ListStatefulsets(ctx, opts.Manifest.Name, ns, c)
-			if err != nil {
-				return err
-			}
-			for i := range sfsList {
-				ss := &sfsList[i]
-				if !statefulsets.IsRunning(ctx, ns, ss.Name, c) {
-					areAllRunning = false
-				}
-			}
-			if !areAllRunning {
-				continue
-			}
-			return nil
-		}
-	}
+	labelSelector := fmt.Sprintf("%s=%s", model.DeployedByLabel, format.ResourceK8sMetaString(opts.Manifest.Name))
+
+	errgroup, ctx := errgroup.WithContext(ctx)
```

Comment:
@ifbyol following up your comment of the initial PR https://github.com/okteto/okteto/pull/4547#pullrequestreview-2413953969 

I've updated the aproach creating 2 errgroups to check the deployments and staefulsets instead, but watching instead of polling.